src/core/core.controller.js

```diff
@@ -67,6 +67,42 @@ module.exports = function(Chart) {
 		return position === 'top' || position === 'bottom';
 	}
 
+	/**
+	 * Return the y data of the label, if no this label, return null
+	 * @private
+	 * @param {string} label the label in the labels array
+	 * @param {array} dataArray the data in the datasets.
+	 * @return {number} the y data according to the label
+	 */
+	function getY(label, dataArray) {
+		for (var i = 0; i < dataArray.length; i++) {
+			var item = dataArray[i];
+			if (item.x === label) {
+				return item.y;
+			}
+		}
+		return null;
+	}
+	/**
+	 * Find if the data in datasets exists in the labels
+	 * @private
+	 * @param {array} dataArray the data in the datasets
+	 * @param {array} labelArray the label in the labels array
+	 * @return {boolean} the match result, true is included
+	 */
+	function dataInLabel(dataArray, labelArray) {
+		for (var i = 0; i < dataArray.length; i++) {
+			var item = dataArray[i];
+			if (item === null || item === undefined || item.x === undefined) {
+				break;
+			}
+			if (labelArray.indexOf(item.x) > -1) {
```

Comment:
this will return true if a single item matches. is that what you want?

if you want to see if all items match then you could return false if < 0 here and return true at the end of the function